CSHTML5.Telerik.Charts/CSHTML5.Extensions.TelerikUI.Controls.Chart/Telerik.Windows.Controls.Chart/Telerik.Windows.Controls/RadPieChart.cs

```diff
@@ -20,79 +22,106 @@ namespace Telerik.Windows.Controls
+                    seriesItem.field = valueMapping.FieldName;
 
-            // add serie to series array
-            series.Add(seriesItem);
+                    // unmapped detail fields
+                    DataPropertyMapping colorMapping = new DataPropertyMapping() { PropName = pieSeries.ColorBinding?.PropertyPath ?? "Color", FieldName = "color" };
+
+                    var propertyFields = new List<DataPropertyMapping>() { categoryMapping, valueMapping, colorMapping };
+                    var res = PrepareSeriesData(pieSeries.ItemsSource, propertyFields);
+                    seriesItem.data = res;
+
+                    series.Add(seriesItem);
+                }
+            }
 
             // add chart to kendo
             chartO.series = series;
             _kendoChart.setOptions(chartO);
         }
 
-        private JSObject GetSeriesData()
+        private JSObject PrepareSeriesData(System.Collections.IEnumerable seriesData, List<DataPropertyMapping> propertiesToPutInResult)
         {
             object preparedSeriesData = Interop.ExecuteJavaScript("[]");
 
-            Interop.ExecuteJavaScript("$0.push($1)", preparedSeriesData, CreateNewPieSeriesObject("Asia", 53.8, "#9de219"));
-            Interop.ExecuteJavaScript("$0.push($1)", preparedSeriesData, CreateNewPieSeriesObject("Europe", 16.1, "#90cc38"));
-            Interop.ExecuteJavaScript("$0.push($1)", preparedSeriesData, CreateNewPieSeriesObject("Latin America", 11.3, "#068c35"));
-            Interop.ExecuteJavaScript("$0.push($1)", preparedSeriesData, CreateNewPieSeriesObject("Africa", 9.6, "#006634"));
-            Interop.ExecuteJavaScript("$0.push($1)", preparedSeriesData, CreateNewPieSeriesObject("Middle East", 5.2, "#004d38"));
-            Interop.ExecuteJavaScript("$0.push($1)", preparedSeriesData, CreateNewPieSeriesObject("North America", 3.6, "#033939"));
+            foreach (var cSharpItem in seriesData)
+            {
+                var jsObject = Interop.ExecuteJavaScript("new Object()");
+                foreach (DataPropertyMapping property in propertiesToPutInResult)
+                {
+                    string propertyName = property.FieldName;
+                    object propertyValue = String.IsNullOrEmpty(property.PropName) ? Utils.GetNestedPropertyValue(cSharpItem, propertyName) : Utils.GetNestedPropertyValue(cSharpItem, property.PropName);
 
+                    if (propertyValue is DateTime) {
+                        Interop.ExecuteJavaScript(@"$0[$1] = new Date($2)", jsObject, propertyName, propertyValue.ToString()); //We'll simply do this for now, it might need some formatting to be sure Date() will understand the date.
+                    } else if (propertyValue != null) {
+                        Interop.ExecuteJavaScript(@"$0[$1] = $2;", jsObject, propertyName, propertyValue.ToString());
```

Comment:
Avoid calling ExecuteJavascript in a loop. Better to use a StringBuilder in order to build the string and then after the foreach statement, call ExecuteJavascript once

Reply:
Solved and also did some cleanup